portlayer/exec/container.go

```diff
@@ -135,5 +138,29 @@ func (c *Container) Start(ctx context.Context) error {
 		return err
 	}
 
-	return nil
+	waitFunc := func(pc []types.PropertyChange) bool {
+		// guestinfo key that we want to wait for
+		key := fmt.Sprintf("guestinfo..sessions|%s.started", c.ID)
+
+		for _, c := range pc {
+			if c.Op != types.PropertyChangeOpAssign {
+				continue
+			}
+
+			values := c.Val.(types.ArrayOfOptionValue).OptionValue
+			for _, value := range values {
+				// check the status of the key and return true if it's "true"
+				if key == value.GetOptionValue().Key && value.GetOptionValue().Value.(string) == "true" {
```

Comment:
We get potentially valuable error info in the Started field when it's not set to true. It also indicates that the command is never going to start as we've tried, failed, and there's no retry mechanism.

I like the context timeout for systemic failure issues such as vSphere communication breaking down, but the presence of this key with a non-zero value is sufficient to break from this wait, and doubles as error detail.